src/engine/ExecuteUpdate.cpp

```diff
@@ -2,57 +2,61 @@
-                                              std::move(graphUpdate.toInsert_));
-  auto [toDeleteTemplates, localVocabDelete] =
-      ExecuteUpdate::transformTriplesTemplate(vocab, qet.getVariableColumns(),
-                                              std::move(graphUpdate.toDelete_));
+  auto prepareTemplateAndResultContainer =
+      [&vocab, &qet,
+       &res](std::vector<SparqlTripleSimpleWithGraph>&& tripleTemplates) {
+        auto [transformedTripleTemplates, localVocab] =
+            transformTriplesTemplate(vocab, qet.getVariableColumns(),
+                                     std::move(tripleTemplates));
+        std::vector<IdTriple<>> updateTriples;
+        // The maximum result size is size(query result) x num template rows.
+        // The actual result can be smaller if there are template rows with
+        // variables for which a result row does not have a value.
+        updateTriples.reserve(res->idTable().size() *
+                              transformedTripleTemplates.size());
+
+        return std::tuple{std::move(transformedTripleTemplates),
+                          std::move(updateTriples), std::move(localVocab)};
+      };
+
+  auto [toInsertTemplates, toInsert, localVocabInsert] =
+      prepareTemplateAndResultContainer(std::move(graphUpdate.toInsert_));
+  auto [toDeleteTemplates, toDelete, localVocabDelete] =
+      prepareTemplateAndResultContainer(std::move(graphUpdate.toDelete_));
 
-  std::vector<IdTriple<>> toInsert;
-  std::vector<IdTriple<>> toDelete;
-  // The maximum result size is size(query result) x num template rows. The
-  // actual result can be smaller if there are template rows with variables for
-  // which a result row does not have a value.
-  toInsert.reserve(res->idTable().size() * toInsertTemplates.size());
-  toDelete.reserve(res->idTable().size() * toDeleteTemplates.size());
   for (const auto& [pair, range] :
        ExportQueryExecutionTrees::getRowIndices(query._limitOffset, *res)) {
     auto& idTable = pair.idTable_;
-    for (uint64_t i : range) {
-      ExecuteUpdate::computeAndAddQuadsForResultRow(toInsertTemplates, toInsert,
-                                                    idTable, i);
+    for (const uint64_t i : range) {
+      computeAndAddQuadsForResultRow(toInsertTemplates, toInsert, idTable, i);
       cancellationHandle->throwIfCancelled();
 
-      ExecuteUpdate::computeAndAddQuadsForResultRow(toDeleteTemplates, toDelete,
-                                                    idTable, i);
+      computeAndAddQuadsForResultRow(toDeleteTemplates, toDelete, idTable, i);
       cancellationHandle->throwIfCancelled();
     }
   }
 
-  // TODO<qup42> use the actual DeltaTriples object
-  DeltaTriples deltaTriples(index);
-  deltaTriples.insertTriples(cancellationHandle, std::move(toInsert));
   deltaTriples.deleteTriples(cancellationHandle, std::move(toDelete));
+  deltaTriples.insertTriples(cancellationHandle, std::move(toInsert));
```

Comment:
Or, even simpler:
Write two functions:
This one here just returns two vectors of triples (toInsert and toDelete), because that is much easier to test.
The other function (that calls the first function and passes it to the delta triples) is then simple, and doesn't have to have such extensive tests. Then you can check the actual vectors, and not the side effects on the `DeltaTriples` object (which is rather complicated, and you have already tested it).